table/block.h

```diff
@@ -80,10 +88,23 @@ class BlockReadAmpBitmap {
 
     // Every new bit we set will bump this counter
     uint32_t new_useful_bytes = 0;
-    // Index of first bit in mask (start_offset / bytes_per_bit)
-    uint32_t start_bit = start_offset >> bytes_per_bit_pow_;
-    // Index of last bit in mask (end_offset / bytes_per_bit)
-    uint32_t end_bit = end_offset >> bytes_per_bit_pow_;
+    // Index of first bit in mask
+    uint32_t start_bit =
+      (start_offset >> bytes_per_bit_pow_ == last_bit_
+             ? last_bit_
```

Comment:
The reason I clamp it is I need to include the last_bit no matter what rnd_ is, which means, for the last bit, if there is one bit overlapped with the range I would take them into consideration. Because if we do rnd_, the last bit may not represents 32 bytes. And also, it breaks some test cases that is not easy to fix by the randomness of rnd_.